sdk/core/azure-core/inc/context.hpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <assert.h>
+#include <chrono>
+#include <memory>
+//#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <type_traits>
+
+constexpr char not_found = 0;
+
+namespace Azure
+{
+namespace Core
+{
+
+struct ValueBase
+{
+  virtual ~ValueBase() {}
+};
+
+/*
+ *  @brief ContextValue exists as a substitute for variant which isn't available until C++17
+ */
+class ContextValue
+{
+  char active;
+  union {
+    bool b; // if active == 1
+    int i; // if active == 2
```

Comment:
Should we have other integer types like long long and unsigned long long too, or does that make the interface too awkward to use since readers might need to inspect more than one? Alternately, should we have a coerce function that tries multiple interpretiation?